### ISSUE TITLE
how to include types for bootstrap-table when compiling typescript in an ASP.NET Core app

### ISSUE THREAD

**Issue by ghosttie** [wenzhixin/bootstrap-table#5816]
I am writing an ASP.NET Core 5 app and instead of using JS I want to use TypeScript.

I'm using some JS libraries, so in order for the TS to compile it needs definitions for them.

I've set up NPM with devDependencies of the type definitions for the libraries, which downloads them to node_modules, and the TS compiler finds and uses them.

My NPM `package.json` is:
```lang-json
{
	"version": "1.0.0",
	"name": "asp.net",
	"private": true,
	"devDependencies": {
		"@types/bootstrap": "5.0.17",
		"@types/jquery": "3.5.6",
		"bootstrap-table": "^1.18.3",
		"jquery": "3.6.0"
	}
}
```

My `tsconfig.json` is:
```lang-json
{
	"compilerOptions": {
		"noImplicitAny": true,
		"noEmitOnError": true,
		"removeComments": true,
		"sourceMap": true,
		"target": "es5",
		"lib": [
			"DOM",
			"ES5",
			"ES2015"
		]
	},
	"include": [
		"wwwroot/js/**/*.ts"
	],
	"exclude": [
		"node_modules"
	],
	"compileOnSave": true
}
```

This works for JQuery and Bootstrap, but it's not working for bootstrap-table.

My project structure is:
```
root
  wwwroot
    js
      home.ts
  views
    home
      index.cshtml      // imports ~/js/home.js
  node_modules          // the folders in here are automatically maintained by NPM
    @types
      bootstrap
      jquery
    bootstrap-table
  package.json
  tsconfig.json
```

The error when compiling `home.ts` is `(TS) Property 'bootstrapTable' does not exist on type 'JQuery<HTMLElement>'.`

I thought the problem might be caused by the type definitions for bootstrap-table being included in the main library, which means it's not in the @types subdirectory of node_modules.

I've tried forcing the TS compiler to see it by manually specifying typeRoots but there are just a lot of compile errors because it starts seeing all sub-folders as modules that are missing.

**Comment by UtechtDustin** [comment]
I close this issue because it's almost 1 year old and i think you don't still need help, right ?
If you need help, let us know!
We will try to help, but i'm not sure if we can help with that combination of TS and asp.net

**Comment by ghosttie** [comment]
I never figured it out so I still need help

I worked around it by writing all of my client-side code in JS rather than TS but that's not the solution I want

**Comment by UtechtDustin** [comment]
Okay good to know, i will reopen the issue.
But i'm not sure if someone of the core dev's can help with that.
Did you try to ask that on stackoverflow ?

**Comment by ghosttie** [comment]
Yes I asked it [on StackOverflow](https://stackoverflow.com/questions/68606042/how-to-include-types-for-bootstrap-table-when-compiling-typescript-in-an-asp-net) but I didn't get any helpful answers

**Comment by djhvscf** [comment]
I can take a look at this today.
